File: LeetCode-Problems/Easy/Sum-All-Odd-Subarrays.py
# Python
class Solution(object):
    def sumOddLengthSubarrays(self, arr):
        """
        :type arr: List[int]
        :rtype: int
        """
        sum = 0
        
        if (len(arr) == 0 or arr == None):
            return 0
        for x in range(0, len(arr)):
            temp = 0
            for y in range (x, len(arr)):
                temp += arr[y]
                if ((y-x) % 2 == 0):
                    sum += temp
        return sum

File: LeetCode-Problems/Easy/Sum-All-Odd-Subarrays.java
//Java
class Solution {
    public int sumOddLengthSubarrays(int[] arr) {
        int sum = 0;
        if (arr.length == 0 || Objects.isNull(arr.length)){
            return 0;
        }
        for (int i = 0; i < arr.length; i++){
            int temp = 0;
            for (int j = i; j < arr.length; j++){
                temp += arr[j];
                if ((j-i) % 2 == 0){
                    sum += temp;
                }
            }
        }
        return sum;
    }
}
